api/client/lib.py

```diff
@@ -443,6 +443,24 @@ def get_data_series(access_token, api_host, **selection):
         raise Exception(resp.text)
 
 
+# TODO: Add code to handle no query parameters call to v2/available/sources. It is hanging at the moment.
+
+
+def get_source_ranking(access_token, api_host, series):
+    """Given a series, return a list of ranked sources.
+
+    :param access_token: API access token.
+    :param api_host: API host.
+    :param series: Series to calculate source raking for.
+    :return: List of sources that match the series parameters, sorted by rank.
+    """
+    url = '/'.join(['https:', '', api_host, 'v2/available/sources'])
+    headers = {'authorization': 'Bearer ' + access_token}
+    params = dict((k + 's', v) for k, v in iter(list(
+        get_params_from_selection(**series).items())))
+    return get_data(url, headers, params).json()
+
+
 def rank_series_by_source(access_token, api_host, series_list):
```

Comment:
added support for time ranges in  https://github.com/gro-intelligence/api-client/pull/125 
and also separating finding frmo ranking..

Reply:
@nemozen I don't understand this comment.

Reply:
just fyi  that this function was changing to also support time range as part of the ranking, which it didn't before.  merged into this now.